optimum/intel/neural_compressor/configuration.py

```diff
@@ -90,3 +93,201 @@ def _create_distillation_config(config: Union[Dict, DistillationConfig]):
+        seq_len: Optional[int] = None,
+        **kwargs,
+    ):
+        """
+        Args:
+            bits (`int`, defaults to 4):
+                The number of bits to quantize to.
+            sym (`bool`, defaults to `False`):
+                Whether to use symmetric quantization.
+            num_samples (`int`, *optional*):
+                The maximum number of samples composing the calibration dataset.
+            batch_size (`int`, *optional*):
+                The batch size number of calibration dataset.
+            seq_len (`int`, *optional*):
+                The sequence length number of calibration dataset.
+        """
+        self.bits = bits
+        self.sym = sym
+        self.num_samples = num_samples
+        self.batch_size = batch_size
+        self.seq_len = seq_len
+
+
+@dataclass
+class INCWeightQuantizationConfig(INCQuantizationConfigBase):
+    """
+    This is a wrapper class about all possible attributes and features that you can play with a model that has been
+    loaded using `optimum-intel` api for weight-only quantization with neural compressor.
+    Args:
+        bits (`int`, defaults to 4):
+            The number of bits to quantize to.
+        sym (`bool`, defaults to `False`):
+            Whether to use symmetric quantization on the weights.
+        num_samples (`int`, *optional*):
+            The maximum number of samples composing the calibration dataset.
+        batch_size (`int`, *optional*):
+            The batch size number of calibration dataset.
+        seq_len (`int`, *optional*):
+            The sequence length number of calibration dataset.
+        tokenizer (`str`, *optional*):
+            The tokenizer used to process the dataset. You can pass either:
+                - A string, the *model id* of a predefined tokenizer hosted inside a model repo on huggingface.co.
+                    Valid model ids can be located at the root-level, like `bert-base-uncased`, or namespaced under a
+                    user or organization name, like `dbmdz/bert-base-german-cased`.
+                - A path to a *directory* containing vocabulary files required by the tokenizer, for instance saved
+                    using the [`~PreTrainedTokenizer.save_pretrained`] method, e.g., `./my_model_directory/`.
+        dataset (`str or List[str]`, *optional*):
+            The dataset used for weight-only quantization with neural-compressor, the default is "NeelNanda/pile-10k".
+        group_size (`int`, *optional*):
+            The group size to use for quantization. Recommended value is 128 and -1 uses per-column quantization.
+        use_layer_wise(`bool`, *optional*):
+            Enable quantize model per layer. `model_name_or_path` parameter is necessary when used `use_layer_wise`, it indicates the model
+            local folder name or the name on remote model hub.
+        quant_method (`str or INCQuantizationMethod`, defaults of INCQuantizationMethod.RTN):
```

Comment:
I think we can only accept strings for now, can still be added if needed in the future, what do you think ? Would need some uniformization with ov configs as well

```suggestion
        quant_method (`str`, defaults to `"rtn"`):
```